### DISCUSSION TITLE
Password is not stored

### DISCUSSION THREAD

**Discussion by twwwt** [subclipse/subclipse#252] [Q&A]
I'm fighting with Eclipse (4.25) and Subclipse (4.3.3.202012181204) to do a seemingly simply thing: store the password for some SVN repository. It tried most of the answers of [this Stackoverflow question](https://stackoverflow.com/q/28855468/3955765) but none helped. Checking the checkbox in the authentication dialog does essentially have no effect; I'm asked again and again for the password.

<img width="383" alt="Enter_Username_and_Password" src="https://user-images.githubusercontent.com/143208/204328244-c81c4017-23bb-4826-9290-0b728a136e49.png">

Anyone that could help?

If it helps, I'm currently using the JavaHL (JNI) interface because, from time to time, I need to work on my local working copies also on the console using the native SVN client; whose version is 1.14.2. Unfortunately, the SVNKit implementation that is distributed together with Subclipse (Update Site) is essentially outdated (1.10.1) and does not support SVN 1.14 working copies.

**Comment by markphip** [comment]
Whether or not to cache your credentials is determined by the API provider (JavaHL in this case). Based on the screenshot it looks like you are on MacOS? Caching usually works there but maybe Eclipse has been denied access to the MacOS Keychain for some reason?

If you are on Linux, then this is not uncommon. SVN stopped supporting caching of plain text passwords so no longer offers to cache on Linux.

You can download the latest zipped archive of the SVNKit library and install via that mechanism. That said, SVN 1.14 did not introduce a new working copy format. It has remained the same since SVN 1.8. SVNKit does have its own caching options so it may be worth trying though in general I recommend using JavaHL. That is the API we are written against and test with. We only support SVNKit via its implementation of the JavaHL API interface.

**Reply by markphip** [reply]
One thing I will add ... JavaHL shares the same code as the SVN command line. You could try using SVN from the command line to do something that requires credentials. If the command line caches those credentials, then normally JavaHL would use those cached credentials. So that could be a way to get them cached ... if nothing else the command line might spit out some more information about why it is not caching.
